test/run/test_stages.py

```diff
@@ -268,6 +299,41 @@ def load_manifest(manifest_name):
             # cache the downloaded data for the files source
             osb.copy_source_data(self.store, "org.osbuild.files")
 
+    def test_qemu(self):
+        datadir = self.locate_test_data()
+        testdir = os.path.join(datadir, "stages", "qemu")
+
+        checks_path = os.path.join(testdir, "checks.json")
+        checks = {}
+        with open(checks_path) as f:
+            checks = json.load(f)
+
+        for image_name, test_data in checks.items():
+            with self.osbuild as osb, tempfile.TemporaryDirectory(dir="/var/tmp") as outdir:
+                osb.compile_file(os.path.join(testdir, "qemu.json"),
+                                exports=[image_name],
+                                output_dir=outdir)
+
+                tree = os.path.join(outdir, image_name)
+                ip = os.path.join(tree, image_name)
+                assert os.path.exists(ip)
+                assert os.path.isfile(ip)
+
+                qemu_img_run = subprocess.run(
+                    ["qemu-img", "info", "--output=json", ip],
+                    capture_output=True,
+                    check=True,
+                    encoding="utf-8"
+                )
+
+                qemu_img_out = json.loads(qemu_img_run.stdout)
+                self.assertTrue(mapping_is_subset(test_data, qemu_img_out),
+                    f"Test data is not a subset of the qemu-img output: {test_data} not <= {qemu_img_run.stdout}"
```

Comment:
Totally fine for now but it would be cool to show which values are different.

Reply:
I agree.

However, I really didn't want to go into such detail in this PR. If we want to have this kind of information, I would rather use some library that produces such diff. I didn't want to implement it myself and also didn't want to pull in additional dependencies. To be honest, I didn't spend too much time on investigating whether osbuild already depends on some library that does this. If yes, I'll happily rework the check.

Given the size of the test data, I assumed that a developer should be able to reasonably find the difference by inspecting the log message.